src/librustdoc/html/format.rs

```diff
@@ -676,6 +676,25 @@ fn primitive_link(
     Ok(())
 }
 
+crate fn primitive_link_url(
+    prim: clean::PrimitiveType,
+    cx: &Context<'_>,
+    root_path: &str,
+) -> Option<String> {
+    let m = &cx.cache();
+    let def_id = m.primitive_locations.get(&prim)?;
+    let crate_name = ExternalCrate { crate_num: def_id.krate }.name(cx.tcx());
+    if def_id.is_local() {
+        return Some(format!("{}{}/primitive.{}.html", root_path, crate_name, prim.as_sym()));
+    }
+    let path = match m.extern_locations[&def_id.krate] {
+        ExternalLocation::Remote(ref s) => format!("{}/{}", s.trim_end_matches('/'), crate_name),
+        ExternalLocation::Local => format!("{}{}", root_path, crate_name),
+        ExternalLocation::Unknown => format!("{}/{}", crate::DOC_RUST_LANG_ORG_CHANNEL, crate_name),
+    };
+    Some(format!("{}/primitive.{}.html", path, prim.as_sym(),))
+}
```

Comment:
Hmm, I kind of want to wait for https://github.com/rust-lang/rust/pull/87073 so we can just delete this function altogether. Would you be ok with that? I hope to get it merged within the week.